### PR TITLE
Fixed SelfHostDeployer and IISExpressDeployer to use the correct appl…

### PR DESCRIPTION
**Pull request by kichalla**
…ication path

@Tratcher @JunTaoLuo

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.Testing/Deployers/IISExpressDeployer.cs [hunk 1]
```diff
@@ -34,21 +34,23 @@ public override DeploymentResult Deploy()
...
+            var webRootLocation = DeploymentParameters.PublishApplicationBeforeDeployment ? DeploymentParameters.PublishedApplicationRootPath : DeploymentParameters.ApplicationPath;
+
             var uri = TestUriHelper.BuildTestUri(DeploymentParameters.ApplicationBaseUriHint);
             // Launch the host process.
-            var hostExitToken = StartIISExpress(uri);
+            var hostExitToken = StartIISExpress(uri, webRootLocation);
 
             return new DeploymentResult
             {
-                WebRootLocation = DeploymentParameters.ApplicationPath,
```
Comment (kichalla) [inline src/Microsoft.AspNetCore.Server.Testing/Deployers/IISExpressDeployer.cs]: BTW, I saw similar issue with Nginx deployer too. If you feel these changes are good to do I will update Nginx deployer too.

**Comment by kichalla** [comment]
@BrennanConroy

File: src/Microsoft.AspNetCore.Server.Testing/Deployers/IISExpressDeployer.cs [hunk 2]
```diff
@@ -57,7 +59,7 @@ private CancellationToken StartIISExpress(Uri uri)
 
                 DeploymentParameters.ServerConfigTemplateContent =
                     DeploymentParameters.ServerConfigTemplateContent
-                        .Replace("[ApplicationPhysicalPath]", DeploymentParameters.ApplicationPath)
+                        .Replace("[ApplicationPhysicalPath]", webRootLocation)
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.Testing/Deployers/IISExpressDeployer.cs]: This should still be ApplicationPath, not webroot. webroot is for static files.

File: src/Microsoft.AspNetCore.Server.Testing/Deployers/IISExpressDeployer.cs [hunk 3]
```diff
@@ -66,7 +68,7 @@ private CancellationToken StartIISExpress(Uri uri)
             }
 
             var parameters = string.IsNullOrWhiteSpace(DeploymentParameters.ServerConfigLocation) ?
-                            string.Format("/port:{0} /path:\"{1}\" /trace:error", uri.Port, DeploymentParameters.ApplicationPath) :
+                            string.Format("/port:{0} /path:\"{1}\" /trace:error", uri.Port, webRootLocation) :
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.Testing/Deployers/IISExpressDeployer.cs]: This one too

File: src/Microsoft.AspNetCore.Server.Testing/Deployers/IISExpressDeployer.cs [hunk 4]
```diff
@@ -34,21 +34,23 @@ public override DeploymentResult Deploy()
                 DotnetPublish();
             }
 
+            var webRootLocation = DeploymentParameters.PublishApplicationBeforeDeployment ? DeploymentParameters.PublishedApplicationRootPath : DeploymentParameters.ApplicationPath;
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.Testing/Deployers/IISExpressDeployer.cs]: Ah, this is just misnamed. webroot is for static files, not app root.

Comment (kichalla) [inline src/Microsoft.AspNetCore.Server.Testing/Deployers/IISExpressDeployer.cs]: ah, you mean this should be named something like `appRoot` instead and rest of the changes are good?

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.Testing/Deployers/IISExpressDeployer.cs]: If you make it appRoot then StartIISExpress is correct, but DeploymentResult.WebRootLocation is wrong. What's WebRootLocation used for?

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.Testing/Deployers/IISExpressDeployer.cs]: I think WebRootLocation was never renamed when we moved web.config from wwwroot to the parent directory. It looks like it should be renamed to AppRoot

Comment (kichalla) [inline src/Microsoft.AspNetCore.Server.Testing/Deployers/IISExpressDeployer.cs]: - I glanced through the tests and do not see this property being used
- From the documentation here https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNetCore.Server.Testing/Common/DeploymentResult.cs#L18-L21, it clearly doesn't indicate about `wwwroot` but i agree `AppRoot` is a better name

Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.Server.Testing/Deployers/IISExpressDeployer.cs]: To be consistent with what we used as hosting keys, we should call it `contentRoot`

**Comment by kichalla** [comment]
:up: :date:

**Comment by Tratcher** [comment]
:shipit:

**Comment by JunTaoLuo** [comment]
:shipit:

**Comment by kichalla** [comment]
808ee7e96542e0e4c47d532572f4a9ae7febf0a8

**Pull request closed without merging**